### ISSUE TITLE
DSC 7.0.0 keystore password was incorrect

### ISSUE THREAD

**Issue by sebplorenz** [FraunhoferISST/DataspaceConnector#866]
Hi,
I switched to version 7.0.0 and I'm getting the following error during startup:
```
2022-01-27 09:21:49 | /dbt_connector_1 log="2022-01-27 08:21:49,755 main ERROR de.fraunhofer.ids.messaging.core.config.ConfigProducer - KeyStoreManager could not be initialized! [code=(IMSCOE0008), exception=(keystore password was incorrect)]" thread=main severity=ERROR message="de.fraunhofer.ids.messaging.core.config.ConfigProducer - KeyStoreManager could not be initialized! [code=(IMSCOE0008), exception=(keystore password was incorrect)]"
2022-01-27 09:21:49 | /dbt_connector_1 log="2022-01-27 08:21:49,754 main ERROR de.fraunhofer.ids.messaging.core.config.ssl.keystore.KeyStoreManager - keystore password was incorrect" container_id=5169e26a3e10c2e5bb482889d23180ddde756c5e76848a0480bf1eec2042c4c6 thread=main severity=ERROR message="de.fraunhofer.ids.messaging.core.config.ssl.keystore.KeyStoreManager - keystore password was incorrect"
2022-01-27 09:21:49 | /dbt_connector_1 log="2022-01-27 08:21:49,753 main ERROR de.fraunhofer.ids.messaging.core.config.ssl.keystore.KeyStoreManager - Could not find Keystore at system scope, aborting! [code=(IMSCOE0015), exception=(keystore password was incorrect)]" thread=main severity=ERROR message="de.fraunhofer.ids.messaging.core.config.ssl.keystore.KeyStoreManager - Could not find Keystore at system scope, aborting! [code=(IMSCOE0015), exception=(keystore password was incorrect)]"
```

I didn't set or changed any keystore password configuration. I assume it uses the default password from the application.properties.
DSC v. 6.4.0 starts up without problems with the exact same configuration.

**Comment by juliapampus** [comment]
As described in the [docs](https://international-data-spaces-association.github.io/DataspaceConnector/Deployment/DatabaseMigration), configurations are not migrated. That means, on startup, a DSC v7 with activated migration loads the configurations from the `config.json` and `application.properties`.

**Comment by ronjaquensel** [comment]
Hi,

I believe this error occurs due to a bug in the persistence of the configuration, that was only fixed in 6.5.0. When the connector persists its configuration in versions prior to 6.5.0, the passwords for the keystores are always null. When reloading the configuration from the database, the password is therefore incorrect.

The problem can be solved in several ways:
* Enable the database migration as described [here](https://international-data-spaces-association.github.io/DataspaceConnector/Deployment/DatabaseMigration), which will remove all old configurations.
* Set `configuration.force.reload=true` in the `application.properties`, which will force the connector to load the configuration from the file.
* Delete the configurations from the database manually: `TRUNCATE TABLE public.configuration CASCADE;`
* Start the connector with an empty database.

**However, only enabling the database migration ensures that all old data will still be usable. Therefore, it is highly recommended to enable this, if an old database is used with v7!**

**Comment by sebplorenz** [comment]
I tried configuration.force.reload=true and it worked.
